src/test/java/com/google/sps/UtilityTest.java

```diff
@@ -115,4 +117,53 @@ public void nullValue() {
 
     Assert.assertEquals(actual, expected);
   }
+
+  /** Tests for getUsername() function */
+  @Test
+  public void validId() {
+    // ID with found user.
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    int userId = 1;
+    
+    String actual = Utility.getUsername(userId, request);
+    String expected = "shaargtz";
+
+    Assert.assertEquals(actual, expected);
+  }
+
+  @Test
+  public void validIdWithNoUser() {
```

Comment:
A common style practice for naming java unit test functions it to include the method being tested, the test condition, and the expected outcome, so something like: getUsername_validIdWithNoUser_returnsEmptyString()

Doing this makes it really explicitly clear what is being tested, how it's being tested, and what we expect to see. Consider updating your methods to also follow this pattern!

Reply:
It's not technically in the public github Java style guide, so I'll leave it up to you - but it's a good habit to get in :)

Reply:
Updated them all!